### ISSUE TITLE
`jj show -T` always generates diff

### ISSUE THREAD

**Issue by kj** [martinvonz/jj#5036]
## Description

When I use `-T` with `jj show`, the diff is always generated and printed after the template output.

## Steps to Reproduce the Problem

In a jj repository with changes to the working copy.

```sh
jj show -T 'self.empty()'
```

## Expected Behavior

Outputs only `true` or `false`.

## Actual Behavior

Outputs `true` or `false` followed by the entire diff of changes in the working copy.

## Specifications

- Platform: Linux x86_64
- Version: jj 0.24.0-32d2a85539254e9d96f9819072fa5c6ac70dd1e4

**Comment by yuja** [comment]
`jj show` is the command to show metadata plus diffs. Maybe you're looking for `jj log --no-graph`?

**Comment by kj** [comment]
I've been using exactly that in the meantime and it does work, although I assumed that `jj show` when used with a template would not show a diff, just like `jj log` does not show its full output when used with a template.

`jj show` feels more natural to me for showing information from a single revision (`jj show -T` vs `jj log --no-graph -r @ -T`), otherwise I'm not sure what the value is in `jj show` accepting a template if it's just going to be drowned out by the diff output?

https://github.com/martinvonz/jj/pull/3010

**Comment by yuja** [comment]
Well, `-T` isn't a flag to suppress diff, and `jj show` is roughly equivalent to `jj log --no-graph -p`. Maybe we could add `jj show --no-patch`, but I don't know if that's useful.

**Comment by kj** [comment]
If `-T` doesn't suppress the diff, then what is the use of it on `jj show`? It does replace (other than the graph) the regular output of `jj log`, and most CLI tools that provide an option for output formatting behave like that. I feel like I may be misunderstanding something? What was the intent of adding it if not to control output?

**Comment by martinvonz** [comment]
@yuja, could we make the diff part of the template? We have support for that these days (thanks to you), so it seems like we should be able to do that.

**Comment by thoughtpolice** [comment]
> If `-T` doesn't suppress the diff, then what is the use of it on `jj show`? It does replace (other than the graph) the regular output of `jj log`, and most CLI tools that provide an option for output formatting behave like that. I feel like I may be misunderstanding something? What was the intent of adding it if not to control output?

I think it's a historical thing. In the past, we didn't have the ability to include diffs as part of the template (as Martin said), so for `show` it was just a way to control the formatting of the metadata/header/front matter, rather than anything else. For example, the default template shows the attached bookmarks, IDs, author, etc, which are all controlled by the templating system.

I suspect nobody has really asked for it before this PR, because nobody really wanted anything else.

**Comment by yuja** [comment]
> @yuja, could we make the diff part of the template? We have support for that these days (thanks to you), so it seems like we should be able to do that.

That's doable (ignoring word wrapping issues and CLI flags), but it would be surprising if `jj show -T` suppressed diffs but `jj log -T -p` didn't. I would rather remove `jj show -T` if the existence of `-T` is confusing.

**Comment by kj** [comment]
I personally find it confusing as I can't think of any situation I'd want to use a template to change the output of the metadata yet still show the diff. If `jj log -T` is the intended way to get custom output even for a single revision, and if there's really no use case for `-T` on `jj show`, then I wouldn't mind the option being removed from `jj show`.

Although it still makes sense to me as a user that `jj show -T` would be a much more intuitive version of the rather awkward `jj log --no-graph -r @ -T` when you just want to get some specific information about the working copy (I use it in my shell prompt).

**Comment by yuja** [comment]
To me, `jj show` is the command to show me a patch, so I've never considered `-T` would suppress the diff part. (If we had `jj patch <subcommand>` namespace, `jj show` might belong there.)

One possible use case of `jj show -T` is something like `jj show -Temail` to render email-compatible patch, but I've never used `jj show -T` to customize the output.

Another idea is to make `jj show` not render diffs by default, and add `jj show -p` flag to do so.

**Comment by kj** [comment]
Isn't that what `jj diff` is for? I'm still new to Jujutsu, so apologies if I have some fundamental misunderstanding.

**Comment by yuja** [comment]
Yeah, kind of. `jj diff` is a general-purpose tool to show diffs between two revisions. `jj show` is a command to show detailed information about a revision, and to me, it is commit metadata and diffs within that revision.

**Comment by martinvonz** [comment]
I'm fine with removing support for `jj show -T`.

**Comment by kj** [comment]
Honestly coming back to this I still find the reasoning very strange. As mentioned by the previous comment, it comes across as `log` multiple commits, `show` a single commit. If `-T` was removed from `show`, I wouldn't be surprised if you eventually had a feature request for it from someone wondering why `log` has it and `show` doesn't.
